src/main/java/org/mule/extension/http/internal/request/HttpClientReflection.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.http.internal.request;
+
+import static org.mule.extension.http.api.request.HttpSendBodyMode.ALWAYS;
+
+import org.mule.extension.http.api.request.HttpSendBodyMode;
+import org.mule.runtime.http.api.client.HttpClient;
+import org.mule.runtime.http.api.client.auth.HttpAuthentication;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.runtime.http.api.domain.message.response.HttpResponse;
+
+import java.lang.reflect.Method;
+import java.util.concurrent.CompletableFuture;
+
+public class HttpClientReflection {
+
+  public static final String HTTP_REQUEST_OPTIONS_CLASS_NAME =
+      "org.mule.runtime.http.api.client.HttpRequestOptions";
+
+  private Object requestOptionsBuilder;
+  Method sendAsyncMethod;
+  Method buildMethod;
+  Method responseTimeoutMethod;
+  Method followsRedirectMethod;
+  Method authenticationMethod;
+  Method sendBodyAlwaysMethod;
+
+  boolean loaded = true;
+
+  public HttpClientReflection(HttpClient client) {
```

Comment:
This way you're doing the reflection thing once per client. Maybe you can have only one instance of this class and receive the client in the sendAsync

Reply:
done